runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -3327,6 +3327,9 @@ void TR_MultipleCallTargetInliner::recursivelyWalkCallTargetAndPerformAction(TR_
 
    debugTrace(tracer(),"recursivelyWalkingCallTargetAndPerformAction: Considering Target %p. node estimate before = %d maxbcindex = %d",ct,action.getNodeEstimate(),getPolicy()->getInitialBytecodeSize(ct->_calleeMethod, 0, comp()));
 
+   if (canSkipCountingNodes(ct))
```

Comment:
This routine is not only about counting nodes, strictly speaking, i.e. the "recursive action" need not be counting nodes, even though it appears to be, presently. Maybe you want to refactor the code such that a dedicated recursive rountine is used just for counting nodes and we can then add the `canSkipCountingNodes` call in that routine.

Reply:
My initial reasoning for adding the check in this routine was because it was already not going to work with anything other than when the `action` is `generateNodeEstimate` because of the following:

https://github.com/eclipse-openj9/openj9/blob/e3d8d3916daad8240be455829097fd95dc417b1f/runtime/compiler/optimizer/InlinerTempForJ9.cpp#L3328

Therefore, I think the better approach might be to change this function itself to be dedicated to node counting rather than adding a new one and letting the original function remain unused.

Reply:
Sure, that is fine with me.